### DISCUSSION TITLE
Audiosources and file objects

### DISCUSSION THREAD

**Discussion by Sparky54** [pschatzmann/arduino-audio-tools#429] [Q&A] [answered]
Hi, I am looking to pass an MP3 audio file to functions in another library to provide information on its length and position in a song once it's playing the file. However, the audio sources in this library that are passed to the audioplayer seem to just take in inputs on the location of the file to look for, leaving me without a file object for me to be able to pass to the functions in the other library. Is there any way around this?

**Comment by pschatzmann** [comment, marked as answer]
I see 2 possiblilities: 
- create your own subclass of AudioPlayer and implement getStream() method which returns p_input_stream.
- extend the AudioPlayer and submit a pull request.
